docs/static/offline-plugins.asciidoc

```diff
@@ -1,17 +1,98 @@
+<<building-offline-packages-deprecated,here>>.
+
+[[building-offline-packs]]
+[float]
+=== Building Offline Plugin Packs
+
+An _offline plugin pack_ is a compressed file that contains all the plugins your offline Logstash installation requires,
+along with the dependencies for those plugins.
+
+To build an offline plugin pack:
+
+. Make sure all the plugins that you want to package are installed on the staging server and that the staging server can
+access the Internet.
+
+. Run the `bin/logstash-plugin prepare-offline-pack` subcommand to package the plugins and dependencies:
++
+[source, shell]
+-------------------------------------------------------------------------------
+bin/logstash-plugin prepare-offline-pack [PLUGINS] --output OUTPUT
+-------------------------------------------------------------------------------
++
+where:
++
+* `[PLUGINS]` specifies one or more plugins that you want to include in the pack.
+* `OUTPUT` specifies the location where the compressed plugin pack will be written. The default location is
++/LOGSTASH_HOME/logstash-offline-plugins-{logstash_version}.zip+.
+
+Examples:
+
+["source","sh",subs="attributes"]
+-------------------------------------------------------------------------------
+bin/logstash-plugin prepare-offline-pack logstash-input-beats <1>
+bin/logstash-plugin prepare-offline-pack logstash-filter-* <2>
+bin/logstash-plugin prepare-offline-pack logstash-filter-* logstash-input-beats <3>
+-------------------------------------------------------------------------------
+<1> Packages the Beats input plugin and any dependencies.
+<2> Uses a wildcard to package all filter plugins and any dependencies.
+<3> Packages all filter plugins, the Beats input plugin, and any dependencies.
+
+NOTE: Downloading all dependencies for the specified plugins may take some time, depending on the plugins listed.
+
+[[installing-offline-packs]]
+[float]
+=== Installing Offline Plugin Packs
+
+To install an offline plugin pack:
+
+. Move the compressed bundle to the machine where you want to install the plugins.
+
+. Run the `bin/logstash-plugin install` subcommand to install the packaged plugins:
++
+["source","sh",subs="attributes"]
+-------------------------------------------------------------------------------
+bin/logstash-plugin install file:///path/to/logstash-offline-plugins-{logstash_version}.zip
+-------------------------------------------------------------------------------
++
+Where +path/to/logstash-offline-plugins-{logstash_version}.zip+ is the path to the offline plugin pack.
+
+//REVIEWERS: Are there any other options from the `install` command that are also supported for an offline plugins install?
+
+//REVIEWERS: In the --help for the latest snapshot build, it doesn't mention that the  --local is deprecated. Looks like the text for the help needs to be updated. Should I open an issue for this, or is this known?
```

Comment:
You are right `--local`  with the install context should also be deprecated, I've took the liberty to create an issue to display a warning see https://github.com/elastic/logstash/issues/6539